packages/osd-stylelint-plugin-stylelint/src/rules/no_custom_colors/index.ts

```diff
@@ -46,15 +47,18 @@ const ruleFunction = (
 
     const isAutoFixing = Boolean(context.fix);
 
-    postcssRoot.walkDecls((decl: any) => {
+    postcssRoot.walkDecls((decl) => {
       if (!isColorProperty(decl.prop)) {
         return;
       }
 
+      // We know this is a rule, because we discriminate the property in the conditional above. This means we only have 1 choice on its type.
+      const parent = decl.parent as Rule;
```

Comment:
I think this could be a little more idiomatic? Maybe something like:

```suggestion
      const parent = getColorPropertyParent(decl);
      if (!parent) {
        return;
      }
```

Reply:
Agree that the explicit type guard is better generally.